persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageDAOImpl.kt

```diff
@@ -33,8 +34,21 @@ class MessageDAOImpl(private val queries: MessagesQueries) : MessageDAO {
 
     override suspend fun deleteAllMessages() = queries.deleteAllMessages()
 
-    override suspend fun insertMessage(message: MessageEntity) = queries.transaction { insertInDB(message) }
+    override suspend fun insertMessage(message: MessageEntity, shouldUpdateConversationOrder: Boolean) {
+        queries.transaction {
+            val selfUserId = QualifiedIDEntity("9371383d-a161-4428-a748-a87e8d0f00c7", "wire.com")
+            val isMessageComingFromOtherClient = message.senderUserId == selfUserId
+            if (isMessageComingFromOtherClient)
+                conversationsQueries.updateConversationReadDate(message.date, message.conversationId)
+            insertInDB(message)
+            if (shouldUpdateConversationOrder)
+                conversationsQueries.updateConversationModifiedDate(message.date, message.conversationId)
+            if (isMessageComingFromOtherClient)
+                conversationsQueries.updateConversationNotificationsDate(message.date, message.conversationId)
+        }
```

Comment:
I found out a cool feature of SQLDelight that might be better than just doing a transaction:

[SQLDelight group statements](https://cashapp.github.io/sqldelight/multiplatform_sqlite/grouping_statements/).

Reply:
Yes, we discussed it on Saturday with Tommaso and Mohammad, but it seems only works on one table at the same time!

Reply:
Just checked again, seems we have something like this:
```
updateSalaryRecord:
BEGIN;

UPDATE salary
SET salary=?
WHERE title=?;

INSERT INTO salaryChangeNotes(time, updateNote)
VALUES(?, ?);

END;
```
this is awesome! I'll try it.